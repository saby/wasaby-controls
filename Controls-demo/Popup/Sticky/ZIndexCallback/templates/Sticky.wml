<Controls.popupTemplate:Sticky headingCaption="Заголовок"
                               headingFontColorStyle="warning"
                               headingFontSize="xl">
    <ws:bodyContentTemplate>
        <div class="controlsDemo-PopupTemplate__list-item controlsDemo_fixedWidth200 controlsDemo__wrapper">
            Lorem ipsum dolor sit amet, consectetur adipisicing elit. Animi, blanditiis consequatur dolorum eos error
            fuga hic inventore iure labore minima nihil nobis porro provident quaerat quibusdam repudiandae similique?
            Excepturi, voluptatum!
        </div>
    </ws:bodyContentTemplate>
</Controls.popupTemplate:Sticky>
