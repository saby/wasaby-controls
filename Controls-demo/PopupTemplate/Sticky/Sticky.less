.controlsDemo-PopupTemplate__footer {
   font-size: 13px;
   padding: 0 12px;
   color: #aaa;
   border-top: 1px dashed #aaa;
}

.controlsDemo-PopupTemplate__list-item {
   height: 32px;
   display: flex;
   align-items: center;
}
