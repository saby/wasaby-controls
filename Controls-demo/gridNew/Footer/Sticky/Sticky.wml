<div class="controlsDemo__wrapper controlDemo__grid-header-sticky">
    <Controls.scroll:Container attr:class="controlsDemo__height400">
        <Controls.grid:View
                keyProperty="key"
                source="{{_viewSource}}"
                stickyFooter="{{true}}"
                columns="{{_columns}}">
            <ws:footerTemplate>
                <ws:partial template="Controls/grid:FooterTemplate">
                    <div>Контент подвала таблицы</div>
                </ws:partial>
            </ws:footerTemplate>
        </Controls.grid:View>
    </Controls.scroll:Container>
</div>
