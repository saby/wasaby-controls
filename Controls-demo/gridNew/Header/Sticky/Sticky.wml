<div class="controlsDemo__wrapper controlDemo__grid-header-sticky">
    <Controls.scroll:Container attr:class="controlsDemo__height400 controlsDemo__width800px">
        <Controls.grid:View
                backgroundStyle="default"
                keyProperty="key"
                source="{{_viewSource}}"
                header="{{_header}}"
                columns="{{_columns}}"/>
    </Controls.scroll:Container>
</div>
