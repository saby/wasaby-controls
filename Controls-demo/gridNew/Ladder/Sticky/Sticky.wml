<div class="controlsDemo__wrapper">
    <Controls.scroll:Container attr:class="controlsDemo__height400" shadowMode="js">
        <Controls.listDeprecate:Mover name="listMover"/>
        <Controls.grid:View
				name="grid"
                keyProperty="key"
                source="{{_viewSource}}"
                columns="{{_columns}}"
                itemsDragNDrop="{{true}}"
                on:customdragStart="_dragStart()"
                on:customdragEnd="_dragEnd()"
                multiSelectVisibility="visible"
                bind:selectedKeys="_selectedKeys"
                header="{{_header}}"
                resultsPosition="top"
                ladderProperties="{{_ladderProperties}}">
			<ws:footerTemplate>
				<ws:partial template="Controls/grid:FooterTemplate">
					<Controls.list:AddButton caption="Add record" on:click="_beginAdd()" data-qa="controlsDemo-GridLadder__addButton"/>
				</ws:partial>
			</ws:footerTemplate>
		</Controls.grid:View>
    </Controls.scroll:Container>
</div>
