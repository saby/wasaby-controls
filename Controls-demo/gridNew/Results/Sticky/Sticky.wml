<div class="controlsDemo__wrapper controlsDemo_fixedWidth800">
    <Controls.scroll:Container attr:class="controlsDemo__height400">
        <Controls.grid:View
                backgroundStyle="default"
                storeId="ResultsSticky"
                header="{{_header}}"
                resultsPosition="top"
                columns="{{_columns}}"/>
    </Controls.scroll:Container>
</div>
