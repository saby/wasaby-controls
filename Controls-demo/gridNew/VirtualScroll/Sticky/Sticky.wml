<div class="controlsDemo__wrapper controlsDemo_fixedWidth500">
    <span class="controls-text-label controlsDemo__mb1">pageSize 50, остальные опции не заданы, показ с первой страницы</span>
    <Controls-demo.list_new.VirtualScroll.NavigationSwitcher.Index storeId="VirtualScrollSticky" totalCount="{{500}}" pageSize="{{50}}">
        <ws:content>
            <Controls.scroll:Container attr:style="height: 300px;">
                <Controls.grid:View
                        storeId="VirtualScrollSticky1"
                        columns="{{_columns}}"
                        header="{{_header}}"
                        groupProperty="group">
                    <ws:virtualScrollConfig pageSize="{{30}}"/>
                </Controls.grid:View>
            </Controls.scroll:Container>
        </ws:content>
    </Controls-demo.list_new.VirtualScroll.NavigationSwitcher.Index>
</div>
