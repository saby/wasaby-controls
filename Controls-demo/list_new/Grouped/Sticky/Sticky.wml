<div class="controlsDemo__wrapper controlsDemo_fixedWidth350">
    <Controls.scroll:Container attr:style="max-height: 200px" shadowMode="js">
        <Controls.list:View
                keyProperty="key"
                source="{{_viewSource}}"
                stickyHeader="{{true}}"
                groupProperty="brand"/>
    </Controls.scroll:Container>
</div>
