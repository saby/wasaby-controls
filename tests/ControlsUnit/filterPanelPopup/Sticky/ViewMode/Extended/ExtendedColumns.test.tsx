/**
 * @jest-environment jsdom
 */
import { unmountComponentAtNode } from 'react-dom';
import { Sticky } from 'Controls/filterPanelPopup';
import { WasabyEvents } from 'UICore/Events';
import 'ControlsUnit/filterPanelPopup/Sticky/EmptyEditorForTests';
import { getInstance } from 'ControlsUnit/filterPanelPopup/UserEventSetup';
import { getExtendedItems } from 'ControlsUnit/filterPanelPopup/Sticky/ViewMode/Extended/getExtendedItems';
import { screen, render, waitFor } from '@testing-library/react';

function getExtendedBlock(container: HTMLElement): HTMLElement {
    return container.querySelector(
        '.controls-FilterViewPanel__additional-editors'
    );
}

describe(
    'Controls/filterPanelPopup:Sticky отображение и скрытие колонок ' +
        '"Можно отобрать" при выборе параметров в фильтре',
    () => {
        setTestID(117229470);

        let container = null;

        beforeEach(() => {
            container = document.createElement('div');
            WasabyEvents.initInstance(container);
            document.body.appendChild(container);
        });
        afterEach(() => {
            unmountComponentAtNode(container);
            WasabyEvents.destroyInstance(container);
            container.remove();
            container = null;
        });

        it('При выборе фильтра из колонки, записи в колонке смещаются', async () => {
            const filterDescription = getExtendedItems(13);

            const userEvent = getInstance();

            render(
                <Sticky
                    items={filterDescription}
                    historyId={'testHistoryId'}
                />,
                { container }
            );

            await userEvent.click(screen.getByText('extendedFilterCaption-1'));
            await screen.findByText('extendedFilterCaption-6');

            expect(getExtendedBlock(container)).toMatchSnapshot();
        });

        it('При выборе всех фильтров из колонки, правая колонка перемещается влево', async () => {
            const filterDescription = getExtendedItems(2);

            const userEvent = getInstance();

            render(<Sticky items={filterDescription} />, { container });

            await userEvent.click(screen.getByText('extendedFilterCaption-0'));
            await waitFor(() => {
                expect(() => {
                    return screen.getByTestId(
                        'controls-FilterViewPanel__additional_left-column'
                    );
                }).toThrow();
            });
        });
    }
);
