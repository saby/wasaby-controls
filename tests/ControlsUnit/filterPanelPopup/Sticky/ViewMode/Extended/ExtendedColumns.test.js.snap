// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Controls/filterPanelPopup:Sticky отображение и скрытие колонок "Можно отобрать" при выборе параметров в фильтре При выборе фильтра из колонки, записи в колонке смещаются 1`] = `
<div
  class="controls-FilterViewPanel__additional-editors controls-FilterViewPanel__additional-editors-popup controls-FilterViewPanel__additional-editors-in-column"
>
  <div
    class="controls-FilterViewPanel__additional_left-column"
    data-qa="controls-FilterViewPanel__additional_left-column"
  >
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-0"
      tabindex="0"
      title="extendedFilterCaption-0"
    >
      extendedFilterCaption-0
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-2"
      tabindex="0"
      title="extendedFilterCaption-2"
    >
      extendedFilterCaption-2
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-3"
      tabindex="0"
      title="extendedFilterCaption-3"
    >
      extendedFilterCaption-3
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-4"
      tabindex="0"
      title="extendedFilterCaption-4"
    >
      extendedFilterCaption-4
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-5"
      tabindex="0"
      title="extendedFilterCaption-5"
    >
      extendedFilterCaption-5
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-6"
      tabindex="0"
      title="extendedFilterCaption-6"
    >
      extendedFilterCaption-6
    </div>
  </div>
  <div
    class="controls-FilterViewPanel__additional_right-column"
    data-qa="controls-FilterViewPanel__additional_right-column"
  >
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-7"
      tabindex="0"
      title="extendedFilterCaption-7"
    >
      extendedFilterCaption-7
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-8"
      tabindex="0"
      title="extendedFilterCaption-8"
    >
      extendedFilterCaption-8
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-9"
      tabindex="0"
      title="extendedFilterCaption-9"
    >
      extendedFilterCaption-9
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-10"
      tabindex="0"
      title="extendedFilterCaption-10"
    >
      extendedFilterCaption-10
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-11"
      tabindex="0"
      title="extendedFilterCaption-11"
    >
      extendedFilterCaption-11
    </div>
    <div
      class="controls-FilterViewPanel__additional-editor controls-fontsize-m controls-text-filterExtendedItem controls-FilterViewPanel__additional-editor-popup controls-FilterViewPanel__additional-editor-popup-in-column"
      data-qa="FilterViewPanel__additional-editor"
      key="filterPanel-extended-extendedFilter-12"
      tabindex="0"
      title="extendedFilterCaption-12"
    >
      extendedFilterCaption-12
    </div>
  </div>
</div>
`;
